pkg/controller/metrics/metrics.go

```diff
@@ -819,6 +818,54 @@ func readMetricsConfig() (*metricsconfig.MetricsConfig, error) {
 	return config, nil
 }
 
+// GetOptionalClusterTypeLabels reads the MetricsWithClusterTypeLabels from the metrics config and returns the same as a
+// map, as well as a slice of metric labels. The latter is helpful for defining the relevant metrics
+func GetOptionalClusterTypeLabels(mConfig *metricsconfig.MetricsConfig) (map[string]string, []string) {
+	var mapClusterTypeLabelToValue = make(map[string]string)
+	var optionalClusterTypeLabels []string
+	for k, v := range mConfig.MetricsWithClusterTypeLabels {
+		mapClusterTypeLabelToValue[k] = v
+		optionalClusterTypeLabels = append(optionalClusterTypeLabels, k)
```

Comment:
`range` is going to iterate over the map keys in random order, which can differ from one invocation to the next. At some point I suspect it'll bite us if `optionalClusterTypeLabels` is in a nondeterministic order. Suggest sorting it.